content/pt/docs/reference/access-authn-authz/authentication.md

```diff
@@ -0,0 +1,889 @@
+```
+ 
+Especificando as opções `--as` e `--as-group`:
+ 
+```bash
+kubectl drain mynode --as=superman --as-group=system:masters
+```
+ 
+```none
+node/mynode cordoned
+node/mynode drained
+```
+ 
+Para personificar um usuário, grupo ou especificar campos extras, o usuário efetuando a personificação deve possuir a permissão de executar o verbo "impersonate" no tipo de atributo sendo personificado ("user", "group", etc.). Para clusters com o plugin de autorização _RBAC_ habilitados, a seguinte ClusterRole abrange as regras necessárias para definir os cabeçalhos de personificação de usuário e grupo:
+ 
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+ name: impersonator
+rules:
+- apiGroups: [""]
+ resources: ["users", "groups", "serviceaccounts"]
+ verbs: ["impersonate"]
+```
+ 
+Campos extras são avaliados como sub-recursos de um recurso denominado "userextras". Para permitir ao usuário que utilize os cabeçalhos de personificação para o campo extra "scopes", o usuário deve receber a seguinte permissão:
+ 
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+ name: scopes-impersonator
+rules:
+# Pode definir o cabeçalho "Impersonate-Extra-scopes".
+- apiGroups: ["authentication.k8s.io"]
+ resources: ["userextras/scopes"]
+ verbs: ["impersonate"]
+```
+Os valores dos cabeçalhos de personificação podem também ser restringidos ao limitar o conjunto de nomes de recursos (`resourceNames`) que um recurso pode ter.
+ 
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+ name: limited-impersonator
+rules:
+# Pode personificar o usuário "jane.doe@example.com"
+- apiGroups: [""]
+ resources: ["users"]
+ verbs: ["impersonate"]
+ resourceNames: ["jane.doe@example.com"]
+ 
+# Pode assumir os grupos "developers" and "admins"
+- apiGroups: [""]
+ resources: ["groups"]
+ verbs: ["impersonate"]
+ resourceNames: ["developers","admins"]
+ 
+# Pode personificar os campos extras "scopes" com valores "view" e "development"
+- apiGroups: ["authentication.k8s.io"]
+ resources: ["userextras/scopes"]
+ verbs: ["impersonate"]
+ resourceNames: ["view", "development"]
+```
+ 
+## plugins de credenciais client-go
+{{< feature-state for_k8s_version="v1.11" state="beta" >}}
+ 
+Ferramentas como `kubectl` e `kubelet`utilizando-se do `k8s.io/client-go` são capazes de executar um comando externo para receber credenciais de usuário.
+ 
+Esta funcionalidade é direcionada à integração do lado cliente, com protocolos de autenticação não suportados nativamente pelo `k8s.io/client-go` como: LDAP, Kerberos, OAuth2, SAML, etc. O plugin implementa a lógica específica do protocolo e então retorna credenciais opacas para serem utilizadas. Quase todos os casos de usos de plugins de credenciais requerem um componente de lado do servidor com suporte para um [autenticador de token webhook](#Token de autenticação via Webhook) para interpretar o formato das credenciais produzidas pelo plugin cliente.
```

Comment:
the `(#Token de autenticação via Webhook)` should be `(#token-de-autenticação-via-webhook)`